tests/lib_integration_tests.rs

```diff
@@ -180,7 +181,7 @@ mod no_conversion {
             case(vec!["a", "aa"], "^a{1,2}$"),
             case(vec!["aaa", "a", "aa"], "^a{1,3}$"),
             case(vec!["aaaa", "a", "aa"], "^(?:a{1,2}|a{4})$"),
-            case(vec!["a", "aa", "aaa", "aaaa", "aaab"], "^(?:a{3}b|a{1,4})$"),
+            case(vec!["a", "aa", "aaa", "aaaa", "aaab"], "^a{1,4}b?$"),
```

Comment:
This is a false positive now. The originally produced regex was correct, the new one is not. Can you fix that? If not, this test case must be commented out.

Reply:
What's wrong with the new regex?

Reply:
The new regex accidentally matches `ab`, `aab` and `aaaab`. It is supposed to match `aaab` only.

Reply:
Ah, I see. Let me put some work into fixing that. I think I know why this happened.